boards/ti/lp_mspm0g3507/lp_mspm0g3507.dts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2025 Texas Instruments
+ * Copyright (c) 2025 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <ti/mspm0/g/mspm0g3507.dtsi>
+#include <ti/mspm0/g/mspm0g1x0x_g3x0x-pinctrl.dtsi>
+#include <zephyr/dt-bindings/clock/mspm0_clock.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "TI LP_MSPM0G3507/MSPM0G3507";
+	compatible = "ti,mspm0g3507";
+
+	aliases {
+		led0 = &led0;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpiob 22 GPIO_ACTIVE_HIGH>;
+			label = "Blue LED";
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <DT_FREQ_M(80)>;
+};
+
+&ulpclk {
+	clock-frequency = <DT_FREQ_M(40)>;
+	clk-div = <2>;
+};
+
+&mclk {
+	clock-frequency = <DT_FREQ_M(80)>;
+	clock-source = <MSPM0_CLOCK_BUS_SYSPLL2X>;
+};
+
+&flash0 {
+	status = "okay";
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x8000>;
+		};
+		slot0_partition: partition@8000 {
+			label = "image-0";
+			reg = <0x00008000 0xB000>;
+		};
+		slot1_partition: partition@13000 {
+			label = "image-1";
+			reg = <0x00013000 0xB000>;
+		};
+		scratch_partition: partition@15000 {
+			label = "image-scratch";
+			reg = <0x00015000 0x2000>;
+		};
```

Comment:
if using swap mode whereby both slots have same number of sectors and uniform sector size then scratch should not be used